### PR TITLE
Added destructor function to clean up the global memstack

### LINKED ISSUE #17: free the global stack at exit
instead of manually doing it because the last msfree may not be called always

### PR DESCRIPTION
**Pull request by Khhs167**
As requested in #17, the code should now clean up the global memstack before exit.

Currently this only compiles for GCC, so I think some research is due.

### CODE REVIEW

**Comment by Khhs167** [comment]
It should now compile on all compilers that support C preproc macros.

File: src/memstack.c [hunk 1]
```diff
@@ -176,7 +176,7 @@ void mspush(memstack* storage, void* ptr) {
         // TODO: Check if this needs to be done elsewhere
         if(storage->first != NULL) {
         	storage->last->next = new_node;
-            new_node->previous = storage->last;
+		      new_node->previous = storage->last;
```
Comment (hrszpuk) [inline src/memstack.c]: Typo?

**Review by hrszpuk — APPROVED** [review]
LGTM! (small typo, can be ignored tbh)

File: src/memstack.c (hunk 1 above)
Comment (Khhs167) [reply src/memstack.c]: yeah, my bad

**Pull request merged**